### PR TITLE
Feat/hornet inx interface tutorial

### PR DESCRIPTION
**Pull request by adrian-grassl**
# Description of change

Added a single-page tutorial to showcase interaction with the IOTA Node Extension (INX) gRPC interface.

## Links to any relevant issues

No

## Type of change

Documentation Enhancement

## How the change has been tested

Wiki was tested locally

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ x ] I have followed the contribution guidelines for this project
- [ x ] I have performed a self-review of my own code
- [ x ] I have commented my code, particularly in hard-to-understand areas
- [ x ] I have tested the wiki locally and tested that all external links work

### CODE REVIEW

**Review by Dr-Electron — CHANGES_REQUESTED** [review]
I would say and recommend/request the same stuff here as in your other tutorial :)

**Comment by Dr-Electron** [comment]
@iota-community/if-technical-writers

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 1]
```diff
@@ -1,10 +1,7 @@
 # INX Interface Interaction
-- These are examples of how to interact with the IOTA Node Extension ([INX](https://github.com/iotaledger/inx)) Interface.
-- The extension uses the [inx.proto](https://github.com/iotaledger/inx/blob/develop/proto/inx.proto) file to generate the required gRPC client to connect to HORNET.
-- [gRPC](https://grpc.io/) is a modern open source high performance Remote Procedure Call (RPC) framework that can run in any environment
-- The examples below show how to use simple requests and streaming requests
+In this tutorial you will learn how to interact with the IOTA Node Extension ([INX](https://github.com/iotaledger/inx)) Interface via javascript (NodeJS) with both simple requests and streaming requests. The extension uses the [inx.proto](https://github.com/iotaledger/inx/blob/develop/proto/inx.proto) file to generate the required gRPC client to connect to HORNET. [gRPC](https://grpc.io/) is a modern open source high performance Remote Procedure Call (RPC) framework that can run in any environment.
```
Comment (lucas-tortora) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:2]:
```suggestion
In this tutorial, you will learn how to interact with the IOTA Node Extension ([INX](https://github.com/iotaledger/inx)) Interface via javascript (NodeJS) with both simple requests and streaming requests. The extension uses the [inx.proto](https://github.com/iotaledger/inx/blob/develop/proto/inx.proto) file to generate the required gRPC client to connect to HORNET. [gRPC](https://grpc.io/) is a modern open source high performance Remote Procedure Call (RPC) framework that can run in any environment.
```

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 2]
```diff
@@ -1,10 +1,7 @@
 # INX Interface Interaction
```
Comment (lucas-tortora) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:1]:
```suggestion
# INX Interface Interaction

```

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 3]
```diff
@@ -1,10 +1,7 @@
 # INX Interface Interaction
-- These are examples of how to interact with the IOTA Node Extension ([INX](https://github.com/iotaledger/inx)) Interface.
-- The extension uses the [inx.proto](https://github.com/iotaledger/inx/blob/develop/proto/inx.proto) file to generate the required gRPC client to connect to HORNET.
-- [gRPC](https://grpc.io/) is a modern open source high performance Remote Procedure Call (RPC) framework that can run in any environment
-- The examples below show how to use simple requests and streaming requests
+In this tutorial you will learn how to interact with the IOTA Node Extension ([INX](https://github.com/iotaledger/inx)) Interface via javascript (NodeJS) with both simple requests and streaming requests. The extension uses the [inx.proto](https://github.com/iotaledger/inx/blob/develop/proto/inx.proto) file to generate the required gRPC client to connect to HORNET. [gRPC](https://grpc.io/) is a modern open source high performance Remote Procedure Call (RPC) framework that can run in any environment.
 
-### Used Components
+### Prerequisites
```
Comment (lucas-tortora) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:4]:
```suggestion
### Prerequisites

```

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 4]
```diff
@@ -13,7 +10,7 @@
 
 ***
 
-### Preparation Steps
+### Set Up
 
 #### Prepare development environment
```
Comment (lucas-tortora) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:15]:
```suggestion
#### Prepare development environment

```

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 5]
```diff
@@ -92,9 +89,7 @@ module.exports = client;
 
 
 #### Prepare proto file
```
Comment (lucas-tortora) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:91]:
```suggestion
#### Prepare proto file

```

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 6]
```diff
@@ -282,8 +277,8 @@ call.on('status', function (status) {
 ***
 
 
-### Execution Steps
-Execute the created files one by one and check out the console output.
+### Run
```
Comment (lucas-tortora) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:280]:
```suggestion
### Run

```

**Review by lucas-tortora — APPROVED** [review]
Just some silly formatting comments, awesome work :) !

**Review by Dr-Electron — APPROVED** [review]

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 7]
```diff
@@ -0,0 +1,313 @@
+# INX Interface Interaction
+
+In this tutorial, you will learn how to interact with the IOTA Node Extension ([INX](https://github.com/iotaledger/inx)) Interface via javascript (NodeJS) with both simple requests and streaming requests. The extension uses the [inx.proto](https://github.com/iotaledger/inx/blob/develop/proto/inx.proto) file to generate the required gRPC client to connect to HORNET. [gRPC](https://grpc.io/) is a modern open source high performance Remote Procedure Call (RPC) framework that can run in any environment.
```
Comment (aleksei-korolev) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:3]:
```suggestion
In this tutorial, you will learn how to interact with the IOTA Node Extension ([INX](https://github.com/iotaledger/inx)) interface via JavaScript (NodeJS) with both simple requests and streaming requests. The extension uses the [inx.proto](https://github.com/iotaledger/inx/blob/develop/proto/inx.proto) file to generate the required gRPC client to connect to HORNET. [gRPC](https://grpc.io/) is a modern open source high performance Remote Procedure Call (RPC) framework that can run in any environment.
```

"interface" is not part of the name and should not be capitalized. "JavaScript" is a name and should be capitalized by its creator's spelling.

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+#### Prepare development environment
+
+Create a new folder for the tutorial in a location of your choice and navigate to it.
+
+```bash
+mkdir inx-interaction
+cd inx-interaction
+```
+
+Run the Node.js initializer to configure the project.
```
Comment (aleksei-korolev) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:26]:
```suggestion
Run the Node.js initializer to configure the project:

```

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 9]
```diff
@@ -0,0 +1,313 @@
...
+```bash
+npm init --yes
+```
+
+Overwrite the `package.json` file with the following content.
```
Comment (aleksei-korolev) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:31]:
```suggestion
Overwrite the `package.json` file with the following content:

```

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 10]
```diff
@@ -0,0 +1,313 @@
...
+- Development Environment and Libraries
+  - Installed [Node.js](https://nodejs.org/en/)
+
+***
+
+### Set Up
+
+#### Prepare development environment
+
+Create a new folder for the tutorial in a location of your choice and navigate to it.
```
Comment (aleksei-korolev) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:19]:
```suggestion
Create a new folder for the tutorial in a location of your choice and navigate to it:
```

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 11]
```diff
@@ -0,0 +1,313 @@
...
+  },
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "Apache-2.0"
+}
+```
+
+Install new dependencies.
```
Comment (aleksei-korolev) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:52]:
```suggestion
Install new dependencies:

```

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 12]
```diff
@@ -0,0 +1,313 @@
...
+```bash
+npm install
+```
+
+***
+
+
+#### Prepare configuration file
+Create new file `config.js` and add the following code.
```
Comment (aleksei-korolev) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:61]:
```suggestion
#### Prepare configuration file

Create new file `config.js` and add the following code:

```

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 13]
```diff
@@ -0,0 +1,313 @@
...
+}
+```
+
+Install new dependencies.
+```bash
+npm install
+```
+
+***
+
```
Comment (aleksei-korolev) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:58]:
```suggestion
```

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 14]
```diff
@@ -0,0 +1,313 @@
...
+const client = new INX(
+    INX_ADDRESS,
+    grpc.credentials.createInsecure()
+);
+
+module.exports = client;
+```
+
+***
+
```
Comment (aleksei-korolev) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:92]:
```suggestion
```

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 15]
```diff
@@ -0,0 +1,313 @@
...
+#### Prepare proto file
+
+Copy the `inx-proto` file from [here](https://github.com/iotaledger/inx/blob/develop/proto/inx.proto) into your folder. The `inx-proto` file contains the definition of all available endpoints and below you will see a few example of both simple requests and streaming requests.
+
+***
+
+### Prepare and run files for simple requests
+#### Create new file `ReadBlock.js` and add the following code.
+```javascript
```
Comment (aleksei-korolev) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:102]:
```suggestion
### Prepare and run files for simple requests

#### Create new file `ReadBlock.js` and add the following code.

```javascript
```

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 16]
```diff
@@ -0,0 +1,313 @@
...
+
+#### Run `ReadBlock.js`
+```sh
+node ./ReadBlock.js
+```
+
+***
+
+#### Create new file `ReadNodeConfiguration.js` and add the following code.
+```javascript
```
Comment (aleksei-korolev) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:163]:
```suggestion
#### Create new file `ReadNodeConfiguration.js` and add the following code.

```javascript

```

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 17]
```diff
@@ -0,0 +1,313 @@
...
+call.on('error', function (error) {
+    // An error has occurred and the stream has been closed.
+});
+call.on('status', function (status) {
+    // process status
+});
+```
+
+#### Run `ListenToBlocks.js`
+```sh
```
Comment (aleksei-korolev) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:251]:
```suggestion
#### Run `ListenToBlocks.js`

```sh
```

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 18]
```diff
@@ -0,0 +1,313 @@
...
+node ./ListenToBlocks.js
+```
+
+***
+
+#### Create new file `ListenToLatestMilestones.js` and add the following code.
+```javascript
```
Comment (aleksei-korolev) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:258]:
```suggestion
#### Create new file `ListenToLatestMilestones.js` and add the following code.

```javascript
```

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 19]
```diff
@@ -0,0 +1,313 @@
...
+call.on('error', function (error) {
+    // An error has occurred and the stream has been closed.
+});
+call.on('status', function (status) {
+    // process status
+});
+```
+
+#### Run `ListenToLatestMilestones.js`
+```sh
```
Comment (aleksei-korolev) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:279]:
```suggestion
#### Run `ListenToLatestMilestones.js`

```sh
```

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 20]
```diff
@@ -0,0 +1,313 @@
...
+node ./ListenToLatestMilestones.js
+```
+
+***
+
+#### Create new file `ReadUnspentOutputs.js` and add the following code.
+```javascript
```
Comment (aleksei-korolev) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:286]:
```suggestion
#### Create new file `ReadUnspentOutputs.js` and add the following code.

```javascript
```

File: content/tutorials/single-page-tutorials/hornet-inx-interaction.md [hunk 21]
```diff
@@ -0,0 +1,313 @@
...
+call.on('error', function (error) {
+    // An error has occurred and the stream has been closed.
+});
+call.on('status', function (status) {
+    // process status
+});
+```
+
+#### Run `ReadUnspentOutputs.js`
+```sh
```
Comment (aleksei-korolev) [inline content/tutorials/single-page-tutorials/hornet-inx-interaction.md:307]:
```suggestion
#### Run `ReadUnspentOutputs.js`

```sh
```

**Review by aleksei-korolev — CHANGES_REQUESTED** [review]

**Review by aleksei-korolev — APPROVED** [review]
i apruv

**Pull request merged**